### PR TITLE
#29143: Improve sigmoid_accurate accuracy using exp21f

### LINKED ISSUE #29143: Improve sigmoid_Accurate accuracy using exp21f
To check and update if sigmoid_Accurate accuracy can be improved by using exp21f : https://github.com/tenstorrent/tt-metal/pull/25946

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
### Ticket
#26993,  #29143

### Problem description
Update sigmoid accurate with exp21f and check accuracy

### What's changed
- **ULP Data** : 
  - Tested for the entire 16-bit range of bfloat16 (0 through 2¹⁶), covering all possible values. Test passes with ULP 1 except for below mentioned cases
  -  Attached ULP Graphs, TTNN vs Torch, Values causing higher ULP : [Graphs](https://docs.google.com/document/d/1jSSbH0YpiJtgrWaiOrpEkK_lmz5TyilARTx8QhAnJFA/edit?tab=t.0#heading=h.9x0mphhxqsxw)
  - Below listed are the failing cases
    - Main branch: 
      - Max ULP Delta: **8.943099975585938** (Mismatching values : [Link](https://drive.google.com/file/d/15IWZ5rHTwB3GWazn2pAqbNP_4vUSp6RZ/view?usp=drive_link))
      - Input Failing range : (-87.0, 5.53125 )
    - Updated branch
      - Max ULP Delta: **2.077178955078125** (Mismatching values : [Link](https://drive.google.com/file/d/11sNnKfxZYYtiFLtd-1EHHpwo3mP2jc-v/view?usp=drive_link))
      - Input Failing range with ULP>1 : (-86.0, 5.53125 )
- **Edge case behaviour** : 

<google-sheets-html-origin>
**INPUT** | **TORCH RESULT** | **TT RESULT (Main)** | **TT RESULT (Updated)**
-- | -- | -- | --
inf | 1 | 1 | 1
-inf | 0 | 0 | 0
nan | nan | 1 | 1
-0.0 | 0.5 | 0.5078 | 0.5
0 | 0.5 | 0.5078 | 0.5

- Performance analysis
  - Single tile Performance analysis : Updated kernel duration is about 98.8% slower
    - Main branch : [sigmoid_accurate_main_perf.csv](https://github.com/user-attachments/files/22326643/sigmoid_accurate_main_perf.csv)
    - Updated branch : [sigmoid_accurate_updated_perf.csv](https://github.com/user-attachments/files/22327504/sigmoid_accurate_updated_perf.csv)
    - Image
  - 8K tile Performance : Updated kernel duration is about 165.7% slower
    - Main branch : [sigmoid_accurate_main_8K_perf.csv](https://github.com/user-attachments/files/22326646/sigmoid_accurate_main_8K_perf.csv)
    - Updated branch : [sigmid_accurate_updated_8K_perf.csv](https://github.com/user-attachments/files/22327508/sigmid_accurate_updated_8K_perf.csv)
    - Image

### Checklist
- [ ] [Blackhole post-commit tests (watcher enabled)](https://github.com/tenstorrent/tt-metal/actions/runs/17892701243) - failed as in main
- [x] [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/runs/17892700324)
- [x] [Blackhole) Blackhole nightly tests](https://github.com/tenstorrent/tt-metal/actions/runs/17892705322) - PASSED
- [ ] [(Single-card) Device perf regressions](https://github.com/tenstorrent/tt-metal/actions/runs/17892709231) - failed as in main
- [ ] [Single card pipeline](https://github.com/tenstorrent/tt-metal/actions/runs/17892707998) - failed as in main
- [x] [Blackhole post-commit tests](https://github.com/tenstorrent/tt-metal/actions/runs/17892703410) 
- [ ] [T3K Pipeline](https://github.com/tenstorrent/tt-metal/actions/runs/17892711516)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by nmauriceTT** [comment]
Thank you for this improvement. 

~~If I understand correctly, UnaryWithParam calls the approx, non-fast version of exp, which is often less accurate than the fast approx version~~: #27433 
~~I also see that the execution time increases significantly (though, that might be acceptable for an accurate operation).~~

~~Can you try with fast-approx exp ?~~ ~~I'm wondering what the trade-off would be with it. (for this, you can just temporarily force `exp_tile/exp_tile_init` so that both approx and fast_and_approx are true).~~

EDIT: Ignore comment, exp_tile was already using fast-approx.

File: tests/ttnn/unit_tests/operations/eltwise/test_sigmoid_accurate_21f.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import torch
+import ttnn
+from tests.ttnn.utils_for_testing import assert_with_ulp
+
+
+def test_sigmoid_accurate_arange(device):
+    # Generate all possible bit pattersn for bf16
```
Comment (nmauriceTT) [inline tests/ttnn/unit_tests/operations/eltwise/test_sigmoid_accurate_21f.py]: Typo: `pattersn` -> `patterns`.

**Review by nmauriceTT — APPROVED** [review]

**Comment by amahmudTT** [comment]
Please add links to passing post commit tests for relevant arcs

**Review by amahmudTT — APPROVED** [review]

**Review by dchenTT — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
- https://github.com/tenstorrent/tt-metal/actions/runs/17941355926 - passed
- https://github.com/tenstorrent/tt-metal/actions/runs/17941334935 - passed
